tensorpipe/test/channel/channel_test.cc

```diff
@@ -356,36 +310,7 @@ class SendTensorsBothWaysTest : public ClientServerChannelTestCase<TBuffer> {
 
     this->peers_->done(PeerGroup::kClient);
     this->peers_->join(PeerGroup::kClient);
-
-    ctx->join();
   }
 };
 
 CHANNEL_TEST_GENERIC(SendTensorsBothWays);
-
-template <typename TBuffer>
-class ContextIsNotJoinedTest : public ClientServerChannelTestCase<TBuffer> {
-  // Because it's static we must define it out-of-line (until C++-17, where we
-  // can mark this inline).
-  static const std::string kReady;
-
- public:
-  void server(std::shared_ptr<transport::Connection> conn) override {
-    std::shared_ptr<Context<TBuffer>> context =
-        this->helper_->makeContext("server");
-    this->peers_->send(PeerGroup::kClient, kReady);
-    context->createChannel({std::move(conn)}, Endpoint::kListen);
-  }
-
-  void client(std::shared_ptr<transport::Connection> conn) override {
-    std::shared_ptr<Context<TBuffer>> context =
-        this->helper_->makeContext("client");
-    EXPECT_EQ(kReady, this->peers_->recv(PeerGroup::kClient));
-    context->createChannel({std::move(conn)}, Endpoint::kConnect);
-  }
-};
-
-template <typename TBuffer>
-const std::string ContextIsNotJoinedTest<TBuffer>::kReady = "ready";
-
-CHANNEL_TEST_GENERIC(ContextIsNotJoined);
```

Comment:
Sorry for changing my mind, but could we keep this around? It's the only test that triggers the TSAN error on MPT, which I find suspicious and I want one day to investigate it well.

Reply:
I added the test back exactly as it was before.